File: python/Solution657.py
class Solution:
    def judgeCircle(self, moves: str):
        move_table = {
            "L": 0,
            "R": 0,
            "U": 0,
            "D": 0
        }

        for i in moves:
            move_table[i] += 1
        return move_table['L'] == move_table['R'] and move_table['U'] == move_table['D']

if __name__ == '__main__':
    result = Solution().judgeCircle("UD")
    print(result)

File: java/src/Solution657.java
import java.util.HashMap;
import java.util.Map;

public class Solution657 {
    public boolean judgeCircle(String moves) {
        Map<Character, Integer> map = new HashMap<>();
        map.put('U', 0);
        map.put('D', 0);
        map.put('L', 0);
        map.put('R', 0);

        for (int i = 0; i < moves.length(); ++i) {
            int new_value = map.get(moves.charAt(i)) + 1;
            map.put(moves.charAt(i), new_value);
        }
        return map.get('U').equals(map.get('D')) && map.get('L').equals(map.get('R'));
    }
}
